g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/dao/impl/S3ProductDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vip.messages.data.dao.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.vmware.vip.common.constants.ConstantsFile;
+import com.vmware.vip.messages.data.conf.S3Cient;
+import com.vmware.vip.messages.data.conf.S3Config;
+import com.vmware.vip.messages.data.dao.api.IProductDao;
+import com.vmware.vip.messages.data.dao.exception.DataException;
+import com.vmware.vip.messages.data.util.S3Utils;
+
+@Repository
+public class S3ProductDaoImpl implements IProductDao {
+
+   private static Logger logger = LoggerFactory.getLogger(S3ProductDaoImpl.class);
+
+   @Autowired
+   private S3Cient s3Client;
+
+   @Autowired
+   private S3Config config;
+
+   /**
+    * get the compose list from S3 server
+    */
+   @Override
+   public List<String> getComponentList(String productName, String version) throws DataException {
+      List<String> componentList = new ArrayList<String>();
+      String filePathPrefix = S3Utils.genProductVersionS3Path(productName, version);
+      ListObjectsV2Result result =
+            s3Client.getS3Client().listObjectsV2(config.getBucketName(), filePathPrefix);
+      if (result == null) {
+         throw new DataException("Can't find S3 resource from " + productName + "\\" + version);
+      }
+      List<S3ObjectSummary> objects = result.getObjectSummaries();
+      if (objects == null || objects.size() < 1) {
+         throw new DataException("S3 Component list is empty.");
+      }
+      for (S3ObjectSummary s3os : objects) {
+         String resultKey =
+               (s3os.getKey().replace(filePathPrefix, "")).split(S3Utils.S3FILE_SEPARATOR)[0];
+         if (!componentList.contains(resultKey)) {
```

Comment:
How about use set, then convert to List in the end?

Reply:
this need to convert